main_server.py

```
@@ -161,18 +161,51 @@ def get_default_steam_info():
+
+
+def join_sync_connector_threads(timeout: float = 3.0) -> list[str]:
+    alive_threads: list[str] = []
+    wait_timeout = max(0.0, float(timeout))
+
+    for name, thread in _iter_sync_connector_threads():
+        try:
+            thread.join(timeout=wait_timeout)
+            if thread.is_alive():
+                alive_threads.append(name)
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)